src/plugins/data_source/server/types.ts

```diff
@@ -57,6 +58,8 @@ export interface AuthenticationMethod {
   name: string;
   authType: AuthType;
   credentialProvider: DataSourceCredentialsProvider;
+  clientPoolSetup: OpenSearchClientPoolSetup;
+  legacyClientPoolSetup: OpenSearchClientPoolSetup;
```

Comment:
The intention here was to provide flexibility to auth type developer to build their own client pool. Auth type developer can decide what type of cache they can used, they can define size, build their unique cache key.

Reply:
Removed client pool from auth method. We will use existing client pool. Modified generateCacheKey to take suffix, so that if any auth method want to generate unique key they can just pass suffix.